aspnetcore/data/ef-rp/crud.md

```diff
@@ -0,0 +1,213 @@
+
+In the preceding sample:
+
+* The second argument (` "student", // Prefix`) is the prefix uses to look up values. It's not case sensitive.
+* The posted form values are converted to the types in the `Student` model using [model binding](xref:mvc/models/model-binding#how-model-binding-works).
+
+<a id="overpost"></a>
+### Overposting
+
+Using `TryUpdateModel` to update fields with posted values is a security best practice because it prevents overposting. For example, suppose the Student entity includes a `Secret` property that this web page should not update or add:
+
+[!code-csharp[Main](intro/samples/cu/Models/StudentZsecret.cs?name=snippet_Intro&highlight=7)]
+
+Even if the app doesn't have a `Secret` field on the create/update Razor Page, a hacker could set the `Secret` value by overposting. A hacker could use a tool such as Fiddler, or write some JavaScript, to post a `Secret` form value. The original code doesn't limit the fields that the model binder uses when it creates a Student instance.
+
+Whatever value the hacker specified for the `Secret` form field is updated in the DB. The following image shows the Fiddler tool adding the `Secret` field (with the value "OverPost") to the posted form values.
+
+![Fiddler adding Secret field](../ef-mvc/crud/_static/fiddler.png)
+
+The value "OverPost" is successfully added to the `Secret` property of the inserted row. The app designer never intended the `Secret` property to be set with the Create page.
+
+<a name="vm"></a>
+### View model
+
+A view model typically contains a subset of the properties included in the model used by the application. The application model is often called the domain model. The domain model typically contains all the properties required by the corresponding entity in the DB. The view model contains only the properties needed for the UI layer (for example, the Create page). In addition to the view model, some apps use a binding model or input model to pass data between the Razor Pages code-behind class and the browser. Consider the following `Student` view model:
+
+[!code-csharp[Main](intro/samples/cu/Models/StudentVM.cs)]
+
+View models provide an alternative way to prevent overposting. The view model contains only the properties to view (display) or update.
+
+The following code uses the `StudentVM` view model to create a new student:
+
+[!code-csharp[Main](intro/samples/cu/Pages/Students/CreateVM.cshtml.cs?name=snippet_OnPostAsync)]
+
+The [SetValues](https://docs.microsoft.com/ dotnet/api/microsoft.entityframeworkcore.changetracking.propertyvalues.setvalues?view=efcore-2.0#Microsoft_EntityFrameworkCore_ChangeTracking_PropertyValues_SetValues_System_Object_) method sets the values of this object by reading values from another [PropertyValues](https://docs.microsoft.com/ dotnet/api/microsoft.entityframeworkcore.changetracking.propertyvalues) object. The other object must be based on the same type as this object, or a type derived from the type for this object
```

Comment:
I'm not sure what this means: "The other object must be based on the same type as this object, or a type derived from the type for this object." SetValues uses property name matching, so the VM type doesn't need to be related to the model type--it just needs to have properties that match.